red/runtime/nodes/Node.js

```diff
@@ -258,6 +316,28 @@ Node.prototype.error = function(logMessage,msg) {
     }
 };
 
+Node.prototype.setCanSendSuccess = function(val) {
```

Comment:
Why does the node have these functions? 
 - `setCanSendSuccess`
 - `canSendSuccess`
 - `maySendSuccess`
?

The last two do the same thing. But I don't think the node should be responsible for tracking this.

Reply:
Excuse me, currently we need only one of the last two functions.
The reason for the need for these functions is for reduction of propagation overhead of successfull completion of a node.

>(HN): Note: Because processing of send and done is on critical path of node processing, we must take care of reducing their execution overhead on implementation.

If the use of node property is not desirable, we can use a flow level dictionary for tracking this information insted with little lookup overhead.